src/pages/Settings/Settings.js

```diff
@@ -1,146 +1,182 @@
   }
+  if (!isEmpty(availableVersion) && !isEmpty(currentVersion) && (availableVersion !== currentVersion)) updateAvailable = true
 
   return <PrimaryLayout headerProps={{ title: 'HoloPort Settings' }}>
-    <strong style={{ marginTop: '20px' }}>Name</strong>
-    <p>
-      {settings.deviceName}
-    </p>
-
-    <strong>URL</strong>
-    <p>
-      {settings.hostUrl}
-    </p>
-
-    <strong>Network ID</strong>
-    <p>
-      {settings.networkId}
-    </p>
-
-    <strong>Access Port Numbers</strong>
-
-    <form styleName='settings-form' onSubmit={handleSubmit(onSubmit)}>
-      <SettingsFormInput
-        label='Device Admin'
-        name='deviceAdminPort'
-        register={register}
-        errors={errors} />
-
-      <SettingsFormInput
-        label='HC Admin'
-        name='hcAdminPort'
-        register={register}
-        errors={errors} />
-
-      <SettingsFormInput
-        label='HC Network'
-        name='hcNetworkPort'
-        register={register}
-        errors={errors} />
-
-      <SettingsFormInput
-        label='Hosting'
-        name='hostingPort'
-        register={register}
-        errors={errors} />
-
-      <Button variant='primary' wide name='update-settings' value='Submit'>Save Changes</Button>
-    </form>
+    <header styleName='jumbotron-header'>
+      {!isEmpty(settings) && <>
+        <HashAvatar seed={settings.hostPubKey} styleName='avatar-image' />
+        <h2> {settings.hostName ? `${settings.hostName}'s` : 'Your'} HoloPort </h2>
+      </> }
+      {/* TODO: Find out what the below number should link to... */}
+      <Button styleName='header-button'>{presentHash(settings.hostPubKey)}</Button>
+    </header>
+
+    <section styleName='settings-section'>
+      <SettingsTable header='Software Version' updateAvailable={updateAvailable}>
+        {!isEmpty(status) && !isEmpty(status.versionInfo)
+          ? <SettingsRow
+            label={presentHash(currentVersion)}
+            content={availableVersion}
+            showSoftwareUpdateModal={showSoftwareUpdateModal}
+            updateAvailable={updateAvailable}
+            versionTable />
+          : <SettingsRow
+            label='Version Number'
+            content={NOT_AVAILABLE} />
+        }
+      </SettingsTable>
+
+      <SettingsTable header='About this HoloPort' >
+        <SettingsRow
+          label='Device Name'
+          content={!isEmpty(settings) && settings.deviceName ? settings.deviceName : NOT_AVAILABLE} />
+        <SettingsRow
+          label='Network ID'
+          // TODO : Determine desired approach for diplaying full networkId Hash...
+          content={!isEmpty(status) && status.networkId ? presentHash(status.networkId, 14) : NOT_AVAILABLE} />
+      </SettingsTable>
+
+      <SettingsTable header='Access Port Numbers'>
+        {!isEmpty(status) && !isEmpty(status.ports)
+          ? Object.entries(status.ports).map(port => {
+            if (port[0] === '__typename') return
```

Comment:
This returns `undefined` which will upset `React` (think this is why you're getting a warning here in the console). `return null` instead when you're returning something to be rendered by `React`.